lib/foreman/engine.rb

```diff
@@ -351,6 +354,14 @@ def flush_reader(reader)
 
 ## Engine ###########################################################
 
+  def check_processes
+    process_name = formation.keys.first
```

Comment:
Is there something better to check here? Maybe that formation is non-zero length?